File: src/main/java/com/skyspaceeleven/service/MembershipService.java
package com.skyspaceeleven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeleven.domain.Membership;
import com.skyspaceeleven.dto.MembershipDTO;
import com.skyspaceeleven.dto.MembershipSearchDTO;
import com.skyspaceeleven.dto.MembershipPageDTO;
import com.skyspaceeleven.dto.MembershipConvertCriteriaDTO;
import com.skyspaceeleven.service.GenericService;
import com.skyspaceeleven.dto.common.RequestDTO;
import com.skyspaceeleven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface MembershipService extends GenericService<Membership, Integer> {

	List<Membership> findAll();

	ResultDTO addMembership(MembershipDTO membershipDTO, RequestDTO requestDTO);

	ResultDTO updateMembership(MembershipDTO membershipDTO, RequestDTO requestDTO);

    Page<Membership> getAllMemberships(Pageable pageable);

    Page<Membership> getAllMemberships(Specification<Membership> spec, Pageable pageable);

	ResponseEntity<MembershipPageDTO> getMemberships(MembershipSearchDTO membershipSearchDTO);
	
	List<MembershipDTO> convertMembershipsToMembershipDTOs(List<Membership> memberships, MembershipConvertCriteriaDTO convertCriteria);

	MembershipDTO getMembershipDTOById(Integer membershipId);







}

File: skyspaceeleven_ui/src/services/MembershipService.js
import http from "../http-common"; 

class MembershipService {
  getAllMemberships(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/membership/memberships`, searchDTO);
  }

  get(membershipId) {
    return this.getRequest(`/membership/${membershipId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/membership?field=${matchData}`, null);
  }

  addMembership(data) {
    return http.post("/membership/addMembership", data);
  }

  update(data) {
  	return http.post("/membership/updateMembership", data);
  }
  
  uploadImage(data,membershipId) {
  	return http.postForm("/membership/uploadImage/"+membershipId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new MembershipService();
